lib/screens/home/products/widgets/product_widget.dart

```diff
@@ -18,19 +21,25 @@ class ProductWidget extends StatelessWidget {
         mainAxisSize: MainAxisSize.min,
         children: [
           IconButton(
-            onPressed: _editItem,
+            onPressed: () => _editItem(context),
             icon: const Icon(Icons.edit),
           ),
           IconButton(
-            onPressed: _removeItem,
+            onPressed: () => _removeItem(context),
             icon: const Icon(Icons.delete),
           ),
         ],
       ),
     );
   }
 
-  void _removeItem() {}
+  void _removeItem(BuildContext context) {
+    context.read<StxProductsBloc>().removeItem(product);
+  }
 
-  void _editItem() {}
+  void _editItem(BuildContext context) {
+    context.router.push(
+      EditProductRoute(product: product),
+    );
+  }
```

Comment:
good practice to create separate functions
but sometimes it looks redundant if a function contains one line.

You can ignore it.